hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotDiffManager.java

```diff
@@ -0,0 +1,302 @@
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.om.snapshot;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.hdds.client.StandaloneReplicationConfig;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.utils.db.cache.CacheKey;
+import org.apache.hadoop.hdds.utils.db.cache.CacheValue;
+import org.apache.hadoop.ozone.OzoneConfigKeys;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.OmSnapshot;
+import org.apache.hadoop.ozone.om.OmTestManagers;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmVolumeArgs;
+import org.apache.hadoop.ozone.om.helpers.SnapshotInfo;
+import org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse;
+import org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse.JobStatus;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.authentication.client.AuthenticationException;
+import org.apache.ozone.test.GenericTestUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.rocksdb.RocksDBException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationFactor.ONE;
+import static org.apache.hadoop.ozone.om.OmSnapshotManager.DELIMITER;
+
+/**
+ * Tests for {@link SnapshotDiffManager}.
+ */
+public class TestSnapshotDiffManager {
+
+  private static final String VOLUME = "vol";
+  private static final String BUCKET = "bucket";
+
+  private static File metaDir;
+  private static OzoneManager ozoneManager;
+  private static OMMetadataManager omMetadataManager;
+  private static SnapshotDiffManager snapshotDiffManager;
+  private static PersistentMap<String, SnapshotDiffJob> snapDiffJobTable;
+
+  @BeforeAll
+  public static void init() throws AuthenticationException,
+      IOException, RocksDBException {
+    metaDir = GenericTestUtils.getRandomizedTestDir();
```

Comment:
You can use @TempDir annotation to create temp dir for the test.

Reply:
@hemantk-12 I had multiple workflows failing due to `@tempdir` not being able to cleanup properly and I found online that other people had the same experience. 

These are some of the failing workflows. They all belong to a dummy branch I had for cancelling a snapshot diff job.

https://github.com/xBis7/ozone/actions/runs/5147594598/jobs/9268500272#step:6:2565

https://github.com/xBis7/ozone/actions/runs/5147452987/jobs/9267857940#step:6:2675

https://github.com/xBis7/ozone/actions/runs/5143362416/jobs/9258280943#step:6:2562

Reply:
I'm surprised and curious what is causing this failure. `@TempDir` is used at some many other place. Were you trying to cleanup or delete it?

Reply:
> Were you trying to cleanup or delete it?

No, the annotation is supposed to take care of creation and cleanup. Check below,

https://github.com/xBis7/ozone/blob/a575a094515f75efe95b9dfda8c9a8bf1de0eb3a/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotDiffManager.java#L69-L70

https://github.com/xBis7/ozone/blob/0694343dae2a3f0237eae4e30b0e5359dde3da75/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotDiffManager.java#L75-L76